### ISSUE TITLE
(Ragged array) Model multiple multinomial distributions with different numbers of supports (or outcomes) 

### ISSUE THREAD

**Issue by inmybrain** [stan-dev/rstan#526]
#### Summary:
- Is there any good ways to input ragged vectors?
- How to write down stan script for modeling a multiple of multinomial distributions with different numbers of supports (or outcomes).

#### Description:
First of all, I'm a complete novice in Rstan.

My question is, generally speaking, if I can input a series of vectors of different lengths in stan file. 

More specifically, let pi_k be a point in the (n_k)-simplex for k=1, ..., K. Here, n_k's may be different, and K is given by a dataset (so you cannot manually define K vectors in your code). I'd like to declare pi_k's in data statement, and generate multinomial distributions with a single trial based on each pi_k.

In other words, it can be written in R script by

>for(k in 1:K){
>  Y_k <- rmultinom(1, pi_k)
>}

or if using list (what I was doing from my MCMC algorithm),
> pi <- list(pi_1, ..., pi_K)
> Y <- lapply(pi, function(x) rmultinom(1, x))

It would be greatly appreciated if anyone helps me.
#### Additional Information:
I've seen this [post in 2016](https://stackoverflow.com/q/35158362) and Example in Chapter 16.2, but I wonder if there is more advanced ways to deal with this matter.

#### Current Version:
v2.17.1

**Comment by bob-carpenter** [comment]
There is a chapter on ragged data structures in the Stan manual.  

This isn't an issue for RStan so I'm closing it.  Please try the message boards:

http://discourse.mc-stan.org

for general questions about Stan.

**Comment by inmybrain** [comment]
I'm sorry for this hassle, and thanks for your guide.

**Comment by bob-carpenter** [comment]
No worries.  It's easy to move discussions to the appropriate venue.  Every project has different conventions around this kind of thing and we'd much rather have people communicate than not.
